### PR TITLE
Fixed unary minus integer bug and support to take function call as vid

### LINKED ISSUE #563: [nGQL] parser bug  for minus 1
nebula> GO FROM 100 OVER like YIELD like._dst -1 as ret1                                                                                              [ERROR (-7)]: syntax error near `-1'                                                                                                                  nebula> GO FROM 100 OVER like YIELD like._dst - 1 as ret1                                                                                             ========                                                                                                                                              | ret1 |                                                                                                                                              ========                                                                                                                                              |  100 |                                                                                                                                              --------                                                                                                                                              Got 1 rows (Time spent: 1675/2524 us)    

-1 vs. -1 1

### PR DESCRIPTION
**Pull request by dutor**
As title.

Now, you can use function call expression where a vertex id is required, like `INSERT VERTEX player(name, age) VALUES hash("Tim Duncan"): ("Tim Duncan", 42)`
close #563

### CODE REVIEW

File: src/common/filter/Expressions.cpp [hunk 1]
```diff
@@ -138,8 +138,10 @@ Expression::decode(folly::StringPiece buffer) noexcept {
 std::string InputPropertyExpression::toString() const {
     std::string buf;
     buf.reserve(64);
-    buf += "$-.";
-    buf += *prop_;
+    buf += "$-";
+    if (prop_ != nullptr) {
+        buf += *prop_;
```
Comment (laura-ding) [inline src/common/filter/Expressions.cpp]: missing .

File: src/graph/InsertEdgeExecutor.cpp [hunk 2]
```diff
@@ -39,13 +39,67 @@ Status InsertEdgeExecutor::prepare() {
...
     for (auto i = 0u; i < rows_.size(); i++) {
         auto *row = rows_[i];
-        auto src = row->srcid();
-        auto dst = row->dstid();
-        auto rank = row->rank();
+        status = row->srcid()->prepare();
+        if (!status.ok()) {
+            break;
+        }
+        auto v = row->srcid()->eval();
```
Comment (laura-ding) [inline src/graph/InsertEdgeExecutor.cpp]: srcId and dstid 's type has been calibrated in paster

File: src/common/filter/Expressions.cpp (hunk 1 above)
Comment (dutor) [reply src/common/filter/Expressions.cpp]: fixed

**Comment by whitewum** [comment]
remember add to nGQL.md after check in

File: src/parser/test/ParserTest.cpp [hunk 3]
```diff
@@ -403,7 +410,7 @@ TEST(Parser, InsertEdge) {
     {
         GQLParser parser;
         std::string query = "INSERT EDGE NO OVERWRITE transfer(amount, time) "
-                            "VALUES 12345->54321@1537408527:(3.75, 1537408527)";
+                            "VALUES 12345->-54321@1537408527:(3.75, 1537408527)";
         auto result = parser.parse(query);
         ASSERT_TRUE(result.ok()) << result.status();
     }
```
Comment (whitewum) [inline src/parser/test/ParserTest.cpp]: try -1->-2

Comment (dutor) [reply src/parser/test/ParserTest.cpp]: What's your point?

Comment (whitewum) [reply src/parser/test/ParserTest.cpp]: a negtive vid to another negtive vid. recommend to cover this simple situation.

**Comment by laura-ding** [comment]
close #563

File: src/graph/GoExecutor.cpp [hunk 4]
```diff
@@ -208,22 +225,30 @@ Status GoExecutor::setupStarts() {
...
         return Status::OK();
     }
-    // Take one column from the input of the pipe
-    DCHECK(colname_ != nullptr);
-    starts_ = inputs_->getVIDs(*colname_);
+    // This is for the case that the former executor has no YIELD clause.
+    // (Even though we already implicitly add `YIELD edge_type._dst AS id`)
+    // In such case, the input ref could be simply written as `$-'.
+    // TODO(dutor) In future, we should remove the implicitly added YIELD clause.
+    static const std::string defaultCol("id");
```
Comment (CPWstatic) [inline src/graph/GoExecutor.cpp]: How about giving this default in parser, like this:
```
| INPUT_REF {
    auto *label = new std::string("id");
    $$ = new InputPropertyExpression(label);
}
```

File: src/common/filter/Expressions.cpp [hunk 5]
```diff
@@ -138,7 +138,8 @@ Expression::decode(folly::StringPiece buffer) noexcept {
 std::string InputPropertyExpression::toString() const {
     std::string buf;
     buf.reserve(64);
-    buf += "$-.";
+    buf += "$-";
+    buf += ".";
```
Comment (CPWstatic) [inline src/common/filter/Expressions.cpp]: Check out this.

**Comment by dutor** [comment]
Jenkins, go!

**Pull request merged**